torch/CMakeLists.txt

```diff
@@ -302,6 +301,14 @@ if (NOT NO_API AND NOT USE_ROCM)
   target_include_directories(torch PUBLIC
     ${TORCH_SRC_DIR}/csrc/api
     ${TORCH_SRC_DIR}/csrc/api/include)
+
+  option(TORCH_WITH_CEREAL "Build the C++ API with Cereal for serialization support" OFF)
```

Comment:
Can we make this `TORCH_USE_CEREAL` to match the other CMake flags? Also, might be best to put this in the root-level CMake (and cmake/Summary.cmake) so people are aware of it. Generally all the options are defined there and then used in caffe2/, torch/, etc.